src/thread_pool.hpp

```diff
@@ -83,24 +83,10 @@ template <typename T, void (T::*process_func)(int index)>
 class nervana::thread_pool
 {
 public:
-    thread_pool(int thread_count)
+    thread_pool(std::vector<int> thread_affinity_map)
```

Comment:
It think it would be better to accept this as a reference or a r-value reference and from such initialize member `m_thread_affinity_map`.

Reply:
I've read a suggestion that if I pass-by-value  an r-value using std::move it still moves instead of copying.

Reply:
the thing I did moves every time I do that instead of doing final copy. i wold have to use forwarding which makes it more messy than its worth it to optimize this when its not critical path.